packages/server/src/fhir/routes.ts

```diff
@@ -117,6 +118,9 @@ protectedRoutes.use('/bulkdata', bulkDataRouter);
 // Async Job
 protectedRoutes.use('/job', jobRouter);
 
+// Validate data before updating ClientApplication
+protectedRoutes.put('/ClientApplication/:id', saveClientValidator);
```

Comment:
I'm thinking this probably should not be directly in an `express` route, should probably be done at the `FhirRouter` level potentially? What do you think, @mattwiller?

I know we have talked about doing something like this across the board long term (having specific validation on a per resource basis, maybe via some "prewrite" pass

That's definitely out of scope for this PR but would be nice to move in a more uniform direction

Reply:
I would actually propose moving this validation into the field definition itself, using a FHIRPath condition like this:
```json
 {
  "id" : "ClientApplication.refreshTokenLifetime",
  "path" : "ClientApplication.refreshTokenLifetime",
  "definition" : "Optional configuration to set the refresh token duration",
  "min" : 0,
  "max" : "1",
  "type" : [{
    "code" : "string"
  }],
  "constraint" : [{
    "key" : "clapp-1",
    "severity" : "error",
    "human" : "Token lifetime must be a valid string representing time duration (eg. 2w, 1h)",
    "expression" : "$this.matches('^[0-9]+[smhdwy]$')"
  }],
  "base": {
    "path" : "ClientApplication.refreshTokenLifetime",
    "min" : 0,
    "max" : "1"
  }
}
```
That way, this validation gets performed at the `Repository` level, any time the resource is updated anywhere in the system.

Reply:
Yeah I actually agree, this is probably the right way to do it